pyhive/hive.py

```diff
@@ -94,12 +96,24 @@ def connect(*args, **kwargs):
     return Connection(*args, **kwargs)
 
 
+# TODO
+# Setting the Cookie in the headers should be implemented in the thrift library.
+# We'll keep this here until that change is available in there.
+class TCookieHttpClient(thrift.transport.THttpClient.THttpClient):
```

Comment:
In the meantime, this commit was merged into Thrift master branch, we just have to wait for a new Thrift release to get rid of this TCookieHttpClient:

https://github.com/apache/thrift/commit/69642f389a06f5ba1b374de52c6b0e29892035d8

Reply:
It looks like this fix has been included in the 0.13.0 Release